File: problems/1200/solution.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference1(vector<int>& arr) {
        int minDif = INT_MAX;
        vector<vector<int>> ans;
        
        sort(begin(arr), end(arr));
        
        for (int i = 1; i < size(arr); i++) {
            int curDif = arr[i] - arr[i - 1];
            
            if (curDif == minDif) {
                ans.push_back({arr[i - 1], arr[i]});
            } else if (curDif < minDif) {
                minDif = curDif;
                ans = {{arr[i - 1], arr[i]}};
            }
        }
        
        return ans;
    }
    
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        int shift = pow(10, 6),
            minEl = arr.front(),
            maxEl = arr.front();
        vector<uint8_t> line(shift * 2 + 1);
        vector<vector<int>> ans;
        
        for (const int& num: arr) {
            line[num + shift] = 1;
            minEl = min(minEl, num);
            maxEl = max(maxEl, num);
        }
        
        for (int prev = minEl + shift, diff = maxEl - minEl, i = prev + 1; i <= maxEl + shift; i++) {
            if (line[i] == 0) {
                continue;
            }
            
            if (i - prev == diff) {
                ans.push_back({prev - shift, i - shift});
            } else if (i - prev < diff) {
                diff = i - prev;
                ans = {{prev - shift, i - shift}};
            }
            
            prev = i;
        }
        
        return ans;
    }
};

File: problems/1200/solution.go
package main
import "sort"

func minimumAbsDifference(arr []int) [][]int {
    sort.Ints(arr)
    res := [][]int{}
    
    minDiff := int(^uint(0) >> 1)
    
    for i := 0; i + 1 < len(arr); i++ {
        if arr[i + 1] - arr[i] < minDiff {
            minDiff = arr[i + 1] - arr[i]
        }
    }
    
    for i := 0; i + 1 < len(arr); i++ {
        if arr[i + 1] - arr[i] == minDiff {
            res = append(res, []int{arr[i], arr[i + 1]})
        }
    }
    
    return res
}